elastic/kibana
# [Deployment Management] Add cards navigation in management landing page for serverless

Fixes https://github.com/elastic/kibana/issues/158230

## Summary

For serverless we want to have a new landing page for the management plugin that links to other specific apps. The management plugin already holds a registry of all the apps that will be shown under its umbrella in the side navigation, so these changes leverage that in order to build a list of categories and apps for each category (that are enabled of course) and enhance them with specific icons and copy we want to have for them in serverless.

This PR does 3 small things:
* Creates a new package that holds the cards navigation component that will be used in the landing page
* Creates a new feature called `CardsNavigation` in the management plugin and exposes an API that can be used by consumers to enable this feature and disable any apps that we dont want to show
* Configures the serverless_observability and serverless_search solutions to use the new navigation. Security solution will use their own navigation.


If certain serverless solutions need to hide specific apps they can do so by using the `hideLinksTo` prop:

```typescript
    management.setupCardsNavigation({
      enabled: true,
      hideLinksTo: ['ingest_pipelines', 'api_keys']
    });
```


##### How to test
* You can manually run `yarn storybook management` to lift up storybook and see the component.
* Kibana
  *  Start elasticsearch with `yarn es snapshot --license=trial` and kibana with `yarn start`, visit http://localhost:5601/uby/app/management/ and verify that we still have the old landing page
  * Start elasticsearch with `yarn es snapshot --license=trial` and kibana with `yarn serverless-oblt`, visit http://localhost:5601/uby/app/management/ and verify that we have the new landing page.

<details>
<summary>Screenshots</summary>

![Screenshot 2023-06-23 at 10 52 53](https://github.com/elastic/kibana/assets/1191206/dc346516-fec9-4917-826e-be968d894072)

![Screenshot 2023-06-29 at 10 04 07](https://github.com/elastic/kibana/assets/1191206/540cef03-5170-409b-9d42-b60ae6be44b5)

![Screenshot 2023-06-29 at 10 05 17](https://github.com/elastic/kibana/assets/1191206/0c8ae262-1fde-4b2b-8cfd-d9e842b66a68)

</details>
